src/Transform/ONNX/ONNXHybridTransformPass.cpp

```diff
@@ -101,19 +119,27 @@ struct ONNXHybridTransformPass
       getDecomposeONNXToONNXPatterns(cumulativePatterns);
     }
 
+    if (convOpt) {
+      getConvOptONNXToONNXPatterns(simdDataLayout, cumulativePatterns);
+    }
+
     patterns = FrozenRewritePatternSet(std::move(cumulativePatterns));
     return success();
   }
 
   void runOnOperation() override {
-    // TODO: check if it's necessary to skip functions with names not
-    // ending in "main_graph" (see ShapeInferencePass.cpp)
     func::FuncOp f = getOperation();
     Region &body = f.getBody();
 
     GreedyRewriteConfig config;
     config.useTopDownTraversal = true;
-    (void)applyPatternsAndFoldGreedily(body, patterns, config);
+    config.maxNumRewrites =
+        maxNumRewrites == -1 ? GreedyRewriteConfig::kNoLimit : maxNumRewrites;
+    if (failed(applyPatternsAndFoldGreedily(body, patterns, config))) {
+      llvm::errs() << "Warning: onnx-hybrid-transform didn't converge with "
+                      "max-num-rewrites="
+                   << maxNumRewrites.getValue() << "\n";
```

Comment:
How did you know that it failed because of "not converge" but other errors like wrong patterns?

Reply:
The applyPatternsAndFoldGreedily returns false only if there are still patterns to apply when the rewrite limit is reached.

Reply:
Thanks for the explanation! I found `applyPatternsAndFoldGreedily` internally emits the same error message using  LLVM_DEBUG: https://github.com/llvm/llvm-project/blob/main/mlir/lib/Transforms/Utils/GreedyPatternRewriteDriver.cpp#L731